libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Core.hs

```diff
@@ -231,7 +266,20 @@ checkConformance implResTest agdaResTest = do
         , "\t\ESC[91m-Implementation"
         , "\t\ESC[92m+Specification\ESC[39m"
         ]
-  unless (implResTest == agdaResTest) $ expectationFailure failMsg
+  unless (implResTest == agdaResTest) $ do
+    mbyCborDumpPath <- liftIO $ lookupEnv "CONFORMANCE_CBOR_DUMP_PATH"
```

Comment:
`lookupEnv` is already lifted. Also avoid duplication of variable names like that, since it is easy for them to go out of sync (see my suggestion below)
```suggestion
    let envVarName = "CONFORMANCE_CBOR_DUMP_PATH"
    mbyCborDumpPath <- lookupEnv envVarName
```